x-pack/solutions/security/plugins/security_solution/public/detections/components/attacks/table/default_group_title_renderers/use_get_default_group_title_renderers.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import { isArray } from 'lodash/fp';
+import type { GroupPanelRenderer } from '@kbn/grouping/src';
+import { useAssistantContext } from '@kbn/elastic-assistant';
+
+import { ALERT_ATTACK_IDS } from '../../../../../../common/field_maps/field_names';
+import { useFindAttackDiscoveries } from '../../../../../attack_discovery/pages/use_find_attack_discoveries';
+import type { AlertsGroupingAggregation } from '../../../alerts_table/grouping_settings/types';
+import { AttackGroupContent } from './attack_group_content';
+
+export interface UseGetDefaultGroupTitleRenderersProps {
+  attackIds?: string[];
+}
+
+export const useGetDefaultGroupTitleRenderers = ({
+  attackIds,
+}: UseGetDefaultGroupTitleRenderersProps) => {
+  const { assistantAvailability, http } = useAssistantContext();
+
+  const { data: attacksData, isLoading: isLoadingAttacks } = useFindAttackDiscoveries({
+    ids: attackIds,
+    http,
+    isAssistantEnabled: assistantAvailability.isAssistantEnabled,
+    perPage: Math.max(attackIds?.length ?? 1, 1),
+  });
+
+  const attacks = useMemo(() => {
+    if (isLoadingAttacks || !attacksData?.data.length) {
+      return [];
+    }
+    return attacksData.data;
+  }, [attacksData?.data, isLoadingAttacks]);
```

Comment:
Nit, this could be simplified to (if you want)
```typescript
  const attacks = useMemo(
    () => (isLoadingAttacks || !attacksData?.data.length ? EMPTY_ARRAY : attacksData.data),
    [attacksData?.data, isLoadingAttacks]
  );
```